packages/components/src/components/primitives/PinInput/PinInput.tsx

```diff
@@ -0,0 +1,39 @@
+import React, { FC } from 'react';
+import {
+  Flex,
+  PinInput as ChakraPinInput,
+  PinInputField as ChakraPinInputField,
+  PinInputFieldProps,
+} from '@chakra-ui/react';
+import { PinInputProps, PinInputStaticMembers } from './types';
+
+const renderFieldsByNumOfDigits: (numOfDigits: number, pinInputFieldProps: PinInputFieldProps) => React.ReactNode[] = (
+  numOfDigits,
+  pinInputFieldProps
+) => {
+  const fields: React.ReactNode[] = [];
+  for (let digit = 0; digit < numOfDigits; digit++) {
+    fields.push(<ChakraPinInputField key={digit} {...pinInputFieldProps} />);
+  }
+  return fields;
+};
+
+export const PinInput: FC<PinInputProps> & PinInputStaticMembers = ({
+  children,
+  numOfDigits = 4,
+  placeholder = '0',
+  pinInputFieldProps,
+  ...rest
+}) => {
+  return (
+    <Flex justifyContent="center" columnGap="0.5rem">
```

Comment:
Do we need this `Flex`? Can we apply these styles in the `ChakraPinInput` theme?

Reply:
I thought the same,  setting the styles to `ChakraPinInput` but it is only a wrapper with a context provider, so at the end we need a container to set the styles, my first approach was to use HStack because that one has 8px  by default (exactly what we need) as margin between children, but I ended up using a flex container because I saw HStack is usually used for storybook.